libs/utils/WordPressUtils/src/main/java/org/wordpress/android/util/ImageUtils.java

```diff
@@ -530,19 +525,19 @@ public static String optimizeImage(Context context, String path, int maxImageWid
         String fileName = MediaUtils.getMediaFileName(file, mimeType);
         String fileExtension = MimeTypeMap.getFileExtensionFromUrl(fileName).toLowerCase();
 
-        int selectedWidth = getImageSize(srcImageUri, context)[0];
-        if (selectedWidth == 0) {
+        int selectedMaxSize = getImageSize(srcImageUri, context)[0];
```

Comment:
I noticed that selecting Original Dimensions doesn't work as expected, since we're taking in consideration the width of the picture only here, where is should be
```
int[] imageDimensions = getImageSize(srcImageUri, context);
int selectedMaxSize = Math.max(imageDimensions[0], imageDimensions[1]);
```